git/object_scanner.go

```diff
@@ -170,22 +127,6 @@ func (s *ObjectScanner) Err() error { return s.err }
 // necessary) and clearing both the object and error fields on the
 // `*ObjectScanner` instance.
 func (s *ObjectScanner) reset() error {
```

Comment:
I think that we might be able to get rid of this function entirely. This used to be used to "advance" the read-head to the end of an object in the event that a caller asked for more before reading the object that they had. I believe that the reason that we used a limited reader is that we copied the remainder of the contents to `ioutil.Discard`, but didn't want to accidentally spill over into the next object.

I think that we used an `*io.LimitedReader` as well here because we wanted to be able to call `.N` to figure out how large the object was. If we remove this function, it should be possible to remove the call to `io.LimitReader`, and instead make the file a pure `io.Reader`.

Reply:
I’ll take a look.  I always love deleting unused code.

Reply:
Yup, looks like both of those things can be cleaned up.